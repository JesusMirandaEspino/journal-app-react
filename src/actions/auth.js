import { types } from '../types/types';

export const login = ( uid, displaName ) => (    {
        type: types.login,
        payload: {
            uid,
            displaName
    }
});
