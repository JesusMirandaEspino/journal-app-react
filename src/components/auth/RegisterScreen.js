import React from 'react';
import validator from 'validator';

import { Link } from "react-router-dom";
import { useForm } from '../../hooks/useForm';

export const RegisterScreen = () => {


    const [ formValues, handleInputChange ] = useForm({
        name: 'Jesus',
        email: 'correo@gmail.com',
        password: '123456',
        password2: '123456'
    });


    const { name, email, password, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        
        if( isFormvalid() ){
            console.log( 'Formulario correcto wewe' );
         }
    }


    const isFormvalid = () => {
        
        if( name.trim().length === 0 ){
            console.log( 'Name is required' );
            return false;
        }else if(  !validator.isEmail( email ) ){
            console.log( 'Email is wrong' );
            return false;
        }else if( password !== password2  && password.length < 5 ){
            console.log( 'The password is wrong' );
            return false;
        }

        return true;

    }


    return (
        <div>
            <h1 className="auth__title" >Register</h1>

            <form onSubmit={ handleRegister } >

                <div className="auth__alert_error" >
                
                    <p>Error en el formulario</p>

                </div>

                <label htmlFor="name" >Name</label>
                <input className="auth__input" type="text"  name="name"  placeholder="Name" value={ name  }  onChange={ handleInputChange } />

                <label htmlFor="email" >Email</label>
                <input className="auth__input" type="email"  name="email"  placeholder="email" value={ email }  onChange={ handleInputChange } />

                <label htmlFor="password" >Password</label>
                <input className="auth__input" type="password"  name="password"  placeholder="password" value={ password }  onChange={ handleInputChange } />

                <label htmlFor="confirm" >Confirm</label>
                <input className="auth__input" type="password"  name="confirm"  placeholder="Confirm Password" value={  password2 }  onChange={ handleInputChange } />

                <button type="submit" className="btn btn-primary btn-block mb-5"  > Login </button>


                <hr />
    

                <Link to="/auth/login" className="link" >Already Register?</Link>


            </form>

        </div>
    )
}
