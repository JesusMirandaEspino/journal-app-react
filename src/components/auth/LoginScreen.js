import React from 'react';
import { Link } from "react-router-dom";
import { useForm } from '../../hooks/useForm';
import { useDispatch, useSelector } from 'react-redux';
import { startLoginEmailpassword, startGoogleLogin } from '../../actions/auth'

export const LoginScreen = () => {

    const dispatch = useDispatch();
    const { loading } = useSelector( state => state.ui );

    const [ formValues, handleInputChange ] = useForm({
        email: 'mail@gmail.com',
        password: '123456'
    });

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch( startLoginEmailpassword( email, password ) );
    }


    const handleGoogleLogin = () => {
        dispatch( startGoogleLogin() );
    }


    return (
        <div>
            <h1 className="auth__title" >LoginScreen</h1>

            <form onSubmit={ handleLogin } >

                <label for="email" >E-mail</label>
                <input className="auth__input" type="text"  name="email"  placeholder="correo@dom.com" value={ email } onChange={ handleInputChange } />

                <label for="password" >Password</label>
                <input className="auth__input" type="password"  name="password"  placeholder="password"  value={ password } onChange={ handleInputChange } />

                <button type="submit" className="btn btn-primary btn-block" disabled={ loading } > Login </button>


                <hr />
                
                <div  className="auth__social_network" >
                    <h2>Login With Social Network</h2>

                        <div 
                            className="google-btn"  onClick={   handleGoogleLogin } >
                            <div className="google-icon-wrapper">
                                <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                            </div>
                            <p className="btn-text">
                                <b>Sign in with google</b>
                            </p>
                        </div>
                </div>


                <Link to="/auth/register" className="link" >Create new account</Link>


            </form>

        </div>
    )
}
