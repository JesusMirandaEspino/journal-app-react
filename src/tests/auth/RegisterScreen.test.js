import {  RegisterScreen } from '../../components/auth/RegisterScreen';
import React from 'react';
import {  mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import  configureStore  from 'redux-mock-store';
import  thunk  from 'redux-thunk';

const middleware = [thunk];
const mockstore = configureStore( middleware );

const initialState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};


let store = mockstore( initialState );

const wrapper = mount(

    <Provider store={ store } >
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>

    );

describe( 'Pruebas con <RegisterScreen />', () => {

    test( 'Debe de mostrarse correctamente', () => {


        expect(wrapper).toMatchSnapshot();


    });


});