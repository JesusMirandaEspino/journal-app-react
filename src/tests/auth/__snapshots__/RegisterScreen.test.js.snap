// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <RegisterScreen /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1
    className="auth__title"
  >
    Register
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Jesus"
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value="correo@gmail.com"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="123456"
    />
    <label
      htmlFor="confirm"
    >
      Confirm
    </label>
    <input
      className="auth__input"
      name="confirm"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
       Login 
    </button>
    <hr />
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already Register?
    </a>
  </form>
</div>
`;
